lib/src/js/polymer_base_dart.html

```diff
@@ -91,10 +105,32 @@
         // If you call notifyPath from dart we need to do manual modifications
         // on any JsArray or JsObject instances.
         var parts = this._getPathParts(path);
-        if (parts.length > 1) {
-          var model = this.get(parts.splice(0, parts.length - 1));
-          if (model[parts[0]] != value) model[parts[0]] = value;
+        // if (parts.length > 1) {
+
+        var thisArg =  this;
+        for (var i=0;i<parts.length-1;i++) {
+          if (Array.isArray(thisArg)&&parts[i]==parseInt(parts[i])) {
+            thisArg=Polymer.Collection.get(thisArg).getItem(parseInt(parts[i]));
+          } else {
+            thisArg=thisArg[parts[i]];
+          }
+        }
+        var last = parts[parts.length-1];
+        if (Array.isArray(thisArg)&&last==parseInt(last)) {
+          Polymer.Collection.get(thisArg).setItem(parseInt(last),value);
+          // WARNING : The following instructon is not right. We should convert from key to index but this isn't possible
```

Comment:
Would it be possible for you to add a test demonstrating this case (I am not 100% sure what you are describing here). You can mark it as skipped (pass the `skip: 'some message'` named argument to the test method).